### DISCUSSION TITLE
Breaking release?

### DISCUSSION THREAD

**Discussion by luigifab** [OpenMage/magento-lts#1741] [Ideas]
Hello everybody!

I suggest to prepare a warning for the [README.md](https://github.com/OpenMage/magento-lts/blob/1.9.4.x/README.md) like this:

| :boom: 21.0.0: It's time to level up!<br>We are removing Mysql4 classes (#1470) and ... |
|:---|

I think I can add a check to allow the execution of the [migration script](https://github.com/OpenMage/magento-lts/pull/1470/commits/3fa49df87883ef34cca23661c9f5f358b138e40b) without crash Mysql4 class. So this will allow to run it on OpenMage 19.x and 20.x without the removal of Mysql4 class. Edit: done.

Yes/No? 🎅🏽

**Comment by fballiano** [comment]
I totally did this idea.

Also, what if the new major is called 22 as it's gonna be 2022? I know that we've nothing in common with annual releases (we're not a smartphone company) but maybe would help to make a new release feel more "modern"?

**Reply by Flyingmana** [reply]
that was actually my main idea behind choosing the last version number (20) to have them follow the years. So yes, 22 would also be my candidate for the next version.

**Reply by luigifab** [reply]
ok, why not, and [20]21 will be deprecated in ten days.

**Reply by addison74** [reply]
What about releasing a new version every month or every 3 months? 2022.1, 2022.2, ... At least for me these figures mean something, month and year. When a new version needs to be released because there are important issues to be resolved quickly a specific code can be added in the version. 

Maybe it's time to talk about a maintenance-only option and an innovative one. Simple OpenMage for maintenance (current 1.9) and OpenMage ... for innovative (current 20, which aims to bring new features to the code, to eliminate obsolescence, but where compatibility issues with unmaintained extensions and changes may occur). For a long time OpenMage LTS went parallel to Magento and it was natural for that to happen but now we are on our own.

A rebranding always brings fresh air :))

**Comment by luigifab** [comment]
Instead of creating a new branch/version by year...
Can I suggest to:
- v19: continue as is
- v20: continue as is
- v21: new branch, with "big breaking changes", each releases with a warning, like Debian Testing or Fedora Rawhide or openSUSE Tumbleweed

And for example, for a PR added in v21, after x months, it can be backported to v20.

**Reply by Flyingmana** [reply]
but then you still have breaking changes in v20, they are just delayed by x months.

**Reply by luigifab** [reply]
I'm not sure to understand.

**Reply by tmotyl** [reply]
Im against creating yet another version. Having 2 versions already (and doing 2 rounds of security patches) is already a big amount of work for maintainers. And i see no real benefit of having 3 versions

**Reply by Sdfendor** [reply]
Isn't 20 the "breaking things"-version already? So 21 would be experimental branch then?

**Reply by luigifab** [reply]
Yes 20 is already the breaking things version, but little breaking thing.
21 will be 20 + **big** breaking changes.
